File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/IgnoreCase.py
#!/usr/bin/python
  
## IgnoreCase.py

import re
import sys

input_string = raw_input("Did you work out today?  ")               #(A)

pattern = r'^\s*(y(es)?|(no?))\s*$'                                 #(B)
pattern_yes = r'y(es)?'                                             #(C)
pattern_no = r'n(o)?'                                               #(D)

m = re.search( pattern, input_string, re.IGNORECASE )               #(E)

if ( m != None ):                                                   #(F)
    answer = m.group(1)                                             #(G)
    m1 = re.search( pattern_yes, answer, re.IGNORECASE )            #(H)
    if ( m1 != None ):                                              #(I)
        print "Great, you were good today\n"                        #(J)
        sys.exit(0)                                                 #(K)
    m2 = re.search( pattern_no, answer, re.IGNORECASE )             #(L)
    if ( m2 != None ):                                              #(M)
        print "Uh oh! you were bad today\n"                         #(N)
        sys.exit(0)                                                 #(O)
else:                                                               #(P)
    print "That was not a legal answer\n"                           #(Q)
    sys.exit(1)                                                     #(R)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/IgnoreCase.pl
#!/usr/bin/perl -w
  
## IgnoreCase.pl 

use strict;

print "Did you exercise today:\n";                                  #(A)

my $input_string = <>;                                              #(B)

my ($answer) = ( $input_string =~ /^\s*(y(es)?|(no?))\s*$/i );      #(C)

if (!$answer) {                                                     #(D)
    print "That was not a legal answer\n";                          #(E)
    exit(1);                                                        #(F)
}

if ( $answer =~ /y(es)?/i ) {                                       #(G)
    print "Great, you were good today\n";                           #(H)
    exit 0;                                                         #(I)
}

if ( $answer =~ /n(o)?/i ) {                                        #(J)
    print "Uh oh! you were bad today\n";                            #(K)
    exit 0;                                                         #(L)
}
